### PR TITLE
Add core/codegen package.

### PR DESCRIPTION
**Pull request by ben-clayton**
Package codegen provides an interface for code generation using LLVM.

codegen provides a higher-level and simpler interface for generating AOT or JIT code than using the LLVM go bindings directly.

Co-authored-by: AWoloszyn <awoloszyn@google.com>

### CODE REVIEW

File: core/codegen/arithmetic.go [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+	return value
+}
+
+// Not returns !x. The type of x must be Bool.
+func (b *Builder) Not(x *Value) *Value {
+	assertTypesEqual(x.ty, b.m.Types.Bool)
+	return b.val(b.m.Types.Bool, b.llvm.CreateNot(x.llvm, ""))
+}
+
+// Not returns ~x.
```
Comment (AWoloszyn) [inline core/codegen/arithmetic.go]: I think this was my fault, but this should be "Invert returns ~x"

File: core/codegen/output.go [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+	"unsafe"
+
+	"llvm/bindings/go/llvm"
+)
+
+func init() {
+	llvm.InitializeAllTargetInfos()
+	llvm.InitializeAllTargets()
+	llvm.InitializeAllTargetMCs()
+	// llvm.InitializeAllAsmParsers()
```
Comment (AWoloszyn) [inline core/codegen/output.go]: Do we still need this here?

File: core/codegen/arithmetic.go (hunk 1 above)
Comment (ben-clayton) [reply core/codegen/arithmetic.go]: Done

File: core/codegen/output.go (hunk 2 above)
Comment (ben-clayton) [reply core/codegen/output.go]: Done

**Pull request merged**